assets/translate.js

```diff
@@ -160,3 +160,27 @@ $( window ).on( 'load', function () {
 	appConfig.imageMapLayer = L.imageOverlay( $imageElement.attr( 'src' ), [ [ 0, 0 ], [ $imageElement.height(), $imageElement.width() ] ] );
 	appConfig.imageMapLayer.addTo( imagemap );
 } );
+
+/**
+ * Disable the upload button once it's been clicked.
+ */
+$( function () {
+	var uploadButtonWidget,
+		$uploadButtonElement = $( '#upload-button-widget' );
+	if ( $uploadButtonElement.length === 1 ) {
+		// If there is an upload button, infuse it and modify its submission behaviour.
+		uploadButtonWidget = OO.ui.infuse( $uploadButtonElement );
+		uploadButtonWidget.on( 'click', function () {
+			// Because we want to disable the upload button after its been clicked but before the
+			// form has been submitted, we need to make sure the 'upload' parameter is still sent
+			// (because that's how we distinguish between upload and download; they're both form
+			// submission buttons on the same form). We do this by creating a new hidden element.
+			var $form = uploadButtonWidget.$element.parents( 'form' ),
```

Comment:
You should also be able to get away with sending the button as upload parameter if you disable it *after* submit, no? That would remove the need for a hidden input?

Reply:
Yeah, I thought that'd work, but it didn't in my testing. :-( I think we're usually able to do that (e.g. recently in Event Metrics) without a problem because we don't often rely on the value of the submit button like we do here.

Reply:
Interesting. Okay, I trust your tests, it's just weird; it should be possible to first submit and then disable, but it might be something odd going on with some timing.